src/librbd/object_map/DiffRequest.cc

```diff
@@ -70,6 +64,32 @@ void DiffRequest<I>::load_object_map(
     return;
   }
 
+  // ignore ENOENT with intermediate snapshots since deleted
+  // snaps will get merged with later snapshots
+  m_ignore_enoent = (m_current_snap_id != m_snap_id_start &&
+                     m_current_snap_id != m_snap_id_end);
+
+  if (m_current_snap_id == CEPH_NOSNAP) {
+    m_current_size = m_image_ctx->size;
+  } else {
+    auto snap_it = m_image_ctx->snap_info.find(m_current_snap_id);
+    if (snap_it == m_image_ctx->snap_info.end()) {
+      ldout(cct, 10) << "snapshot " << m_current_snap_id << " does not exist"
+                     << dendl;
+      if (!m_ignore_enoent) {
```

Comment:
@dillaman I am not sure I see how it is going to work for `m_snap_id_start == 0` case. Won't it always return `-ENOENT` for this case?

Reply:
Hmm, yeah -- not sure why that didn't bomb on the unit tests. Anyway, added a condition around adding the start snap to the set.

Reply:
I suppose because the deep_copy just skips the optimization if `object_map::DiffRequest` fails, and it is difficult to catch in the tests. It would be great if we could catch such things in the tests somehow, to avoid the cases as we had recently when the optimization had been always skipped...

Reply:
No, I was wondering why the DiffIterate* set of tests didn't catch it since the librbd API just reuses this if fast-diff is enabled.

Reply:
... answered my own question: same thing, it silently falls back

Reply:
I am recalling we added `m_snap_id_start == 0` for deep-copy (migration?) so `DiffIterate` probably just does not test this. Though I may be remembering wrong.